sonar-python-plugin/src/main/java/org/sonar/plugins/python/flake8/Flake8ReportReader.java

```diff
@@ -19,89 +19,42 @@
  */
 package org.sonar.plugins.python.flake8;
 
-import java.io.File;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Scanner;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import javax.annotation.Nullable;
-import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugins.python.TextReportReader;
 
-public class Flake8ReportReader {
+public class Flake8ReportReader extends TextReportReader {
 
   private static final Logger LOG = Loggers.get(Flake8ReportReader.class);
   private static final Pattern DEFAULT_PATTERN = Pattern.compile("(.+):(\\d+):(\\d+): (\\S+) (.*)");
   private static final Pattern PYLINT_PATTERN = Pattern.compile("(.+):(\\d+): \\[(.*)\\] (.*)");
 
-  public List<Issue> parse(File report, FileSystem fileSystem) throws IOException {
-    List<Issue> issues = new ArrayList<>();
-    try (Scanner scanner = new Scanner(report.toPath(), fileSystem.encoding().name())) {
-      while (scanner.hasNextLine()) {
-        Issue issue = parseLine(scanner.nextLine());
-        if (issue != null) {
-          issues.add(issue);
-        }
-      }
-    }
-    return issues;
-  }
-
-  private static Issue parseLine(String line) {
+  protected Issue parseLine(String line) {
 
     if (line.length() > 0) {
-      if (!startsWithWhitespace(line)) {
-        Matcher m = DEFAULT_PATTERN.matcher(line);
-        if (m.matches()) {
-          String filePath = m.group(1);
-          int lineNumber = Integer.parseInt(m.group(2));
-          int columnNumber = Integer.parseInt(m.group(3));
-          String ruleKey = m.group(4);
-          String message = m.group(5);
-          return new Issue(filePath, ruleKey, message, lineNumber, columnNumber);
-        }
-        m = PYLINT_PATTERN.matcher(line);
-        if (m.matches()) {
-          String filePath = m.group(1);
-          int lineNumber = Integer.parseInt(m.group(2));
-          String ruleKey = m.group(3);
-          String message = m.group(4);
-          return new Issue(filePath, ruleKey, message, lineNumber, null);
-        }
-        LOG.debug("Cannot parse the line: {}", line);
-      } else {
-        LOG.debug("Classifying as detail and ignoring line '{}'", line);
+      Matcher m = DEFAULT_PATTERN.matcher(line);
+      if (m.matches()) {
+        String filePath = m.group(1);
+        int lineNumber = Integer.parseInt(m.group(2));
+        int columnNumber = Integer.parseInt(m.group(3));
+        // Flake8 issues are off by 1 compared to SQ conventions
+        columnNumber -= 1;
+        String ruleKey = m.group(4);
+        String message = m.group(5);
+        return new Issue(filePath, ruleKey, message, lineNumber, columnNumber);
       }
+      m = PYLINT_PATTERN.matcher(line);
```

Comment:
I would add a comment here that justifies why are we using a Pylint pattern here (maybe also worth explaining that even though we have the same pattern in `PylintReportReader`, the implementation is slightly different).

Reply:
After having a second look at this, the different formats are only subtly different and I decided to simply make a more generic version of the regexes and logic in `TextReportReader` rather than keep the 2 implementations.